components/nodemanager-service/pgdb/project_update.go

```diff
@@ -0,0 +1,267 @@
+package pgdb
+
+import (
+	"context"
+
+	"encoding/json"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/components/nodemanager-service/api/nodes"
+	project_update_lib "github.com/chef/automate/lib/authz"
+	"github.com/chef/automate/lib/stringutils"
+	uuid "github.com/chef/automate/lib/uuid4"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const selectNodesProjectData = `
+SELECT
+  n.id,
+  n.projects_data
+FROM nodes n
+`
+
+type NodeProjectData struct {
+	ID                   string `db:"id"`
+	Data                 string `db:"projects_data"`
+	ProjectDataKeyValues []*nodes.ProjectsData
+}
+
+// JobCancel - cancel the a currently running project update
+func (db *DB) JobCancel(ctx context.Context, projectID string) error {
+	log.Debugf("Node manager project update JobCancel %s", projectID)
+	if db.ProjectUpdate.ID == projectID {
+		db.ProjectUpdate.running = false
+	}
+	return nil
+}
+
+// UpdateProjectTags - start a project update
+func (db *DB) UpdateProjectTags(ctx context.Context,
+	projectRules map[string]*iam_v2.ProjectRules) ([]string, error) {
+	jobUUID, _ := uuid.NewV4()
```

Comment:
nitpick: This only ever uses the `.String()` version and repeats the conversion 3 times, so that could be done once here.